core/src/test/scala/org/apache/spark/storage/BlockManagerDecommissionSuite.scala

```diff
@@ -69,36 +84,64 @@ class BlockManagerDecommissionSuite extends SparkFunSuite with LocalSparkContext
     })
 
     // Cache the RDD lazily
-    sleepyRdd.persist()
+    if (persist) {
+      testRdd.persist()
+    }
 
     // Start the computation of RDD - this step will also cache the RDD
-    val asyncCount = sleepyRdd.countAsync()
+    val asyncCount = testRdd.countAsync()
 
     // Wait for the job to have started
     sem.acquire(1)
 
+    // Give Spark a tiny bit to start the tasks after the listener says hello
+    Thread.sleep(100)
```

Comment:
Please consider using `eventually` here, like:
```
 eventually(timeout(300.milliseconds()), interval(10.milliseconds)) {
   assert(sched.getExecutorIds.size == numExecs)
}
```

I believe this is more robust than a constant time for waiting to start the executors.

Then you can remove:
https://github.com/apache/spark/blob/be2a5e736e051ca0497906b2a2e904c7b4033596/core/src/test/scala/org/apache/spark/storage/BlockManagerDecommissionSuite.scala#L103

Reply:
So this wait is for the task to be properly scheduled not for the number of execs. The assert for the number of execs is something I think we should keep because we can to make sure that decommissioning isn’t the same as exiti.

Reply:
I see but please consider to come up with a condition and wait for it to be satisfied instead of waiting for a fixed amount of time. This would make the test more stable and potentially at some runs more quicker.

Reply:
if I think of one I'll update it. There might be another listener event I can look at but I'll revisit this nearer the end.

Reply:
I've added the wait for all the executors to come up before starting the job, but I think this sleep is ok because we know it's less than the length of the job and we are essentially trying to test what happens in the middle of a job. I can't think of a reasonable way to avoid this sleep.